dashboards/grafana-dashboard-assisted-installer-validations.configmap.yaml

```diff
@@ -0,0 +1,288 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: assisted-installer-ocm-operations
+  labels:
+    grafana_dashboard: "true"
+  annotations:
+    grafana-folder: /grafana-dashboard-definitions/AssistedInstaller
+data:
+  assisted-installer.json: |-
+    {
+      "annotations": {
+        "list": [
+          {
+            "builtIn": 1,
+            "datasource": "-- Grafana --",
+            "enable": true,
+            "hide": true,
+            "iconColor": "rgba(0, 211, 255, 1)",
+            "name": "Annotations & Alerts",
+            "type": "dashboard"
+          }
+        ]
+      },
+      "editable": true,
+      "gnetId": null,
+      "graphTooltip": 0,
+      "id": 5,
+      "iteration": 1618385145527,
+      "links": [],
+      "panels": [
+        {
+          "datasource": "$datasource",
+          "fieldConfig": {
+            "defaults": {
+              "custom": {},
+              "mappings": [],
+              "thresholds": {
+                "mode": "absolute",
+                "steps": [
+                  {
+                    "color": "green",
+                    "value": null
+                  },
+                  {
+                    "color": "red",
+                    "value": 80
+                  }
+                ]
+              }
+            },
+            "overrides": []
+          },
+          "gridPos": {
+            "h": 9,
+            "w": 12,
+            "x": 0,
+            "y": 0
+          },
+          "id": 2,
+          "options": {
+            "displayMode": "gradient",
+            "orientation": "horizontal",
+            "reduceOptions": {
+              "calcs": [
+                "mean"
+              ],
+              "fields": "",
+              "values": false
+            },
+            "showUnfilled": true
+          },
+          "pluginVersion": "7.2.1",
+          "targets": [
+            {
+              "expr": "sum(service_assisted_installer_host_validation_failed_after_success_before_installation) by (hostValidationType)",
```

Comment:
I noticed your PromQL queries do not use [rate()](https://prometheus.io/docs/prometheus/latest/querying/functions/#rate) or [increase()](https://prometheus.io/docs/prometheus/latest/querying/functions/#increase), is that intentional?
Did you verify the numbers are correct? specifically, do the numbers make sense over time? (e.g. select 1 hour time range vs. 1 day - and see it matches the raw data in Prometheus)

Some related article on how to use sum and rate: https://www.robustperception.io/rate-then-sum-never-sum-then-rate

Reply:
It is intended.
We have decided to start with the most basic information regarding the last 7 days, the purpose is to currently "mark" the validations that are the highest "trouble makers" and to improve them.

If we will need more sophisticated queries we will add them once we know what additional information we need.

Reply:
Still, it will not handle counter resets, which is what happens every time a pod restarts for any reason (a bug or a new version deployed by @gamli75).

I have compared the numbers for `connected` between Grafana and Prometheus.
I'll attach my findings
![Screen Shot 2021-04-18 at 12 27 12](https://user-images.githubusercontent.com/2920292/115140849-1117f300-a042-11eb-9705-dc269717ac62.png)
![Screen Shot 2021-04-18 at 12 28 37](https://user-images.githubusercontent.com/2920292/115140851-12492000-a042-11eb-835f-e03567d6f712.png)

Reply:
Few questions:
1. Why using `sum` (without `rate` nor `increase`) doesn't reset on pod's restart?! If a pod restart, then its counter restart as well and prometheus should scrape the new value as well no?!
2. I have noticed that when I use the `Graph` visualisation in Grafana, I do get the same raw data but for some reason this is not the case when using the `Bar gauge` visualisation.

Reply:
1. It is how Prometheus counters work. if the pod got restart you'll get new scrapes, but the counting will start from 0, not from where it was last reported (as you see in the Graph). you may find information about this in [rate() Prometheus docs](https://prometheus.io/docs/prometheus/latest/querying/functions/#rate) or [in this blog post](https://www.robustperception.io/how-does-a-prometheus-counter-work).
2. So you should change the displayed value from `Mean` to `Last` for all pannels (and update the exported JSON). note that it fixes things, but not the counter reset
![Screen Shot 2021-04-18 at 17 58 17](https://user-images.githubusercontent.com/2920292/115150148-d0cf6980-a06f-11eb-8af8-7d19e1303cf2.png)

Reply:
1. So it DOES reset the counters, this is exactly the behaviour I want, I don't want to see metrics from an old version of the service. If we solved a bug I want to see fresh, new, metrics not affected by the older version.
2. Will do

Reply:
what if the reset happened for another reason? and the service version remains the same?

Reply:
That's a good point.
Adding `Increase()` to it.